packages/models-library/src/models_library/projects.py

```diff
@@ -1,153 +1,106 @@
+from .projects_nodes import Node
+from .projects_nodes_io import NodeID_AsDictKey
+from .projects_state import ProjectState
+from .projects_ui import StudyUI
 
-current_file = Path(sys.argv[0] if __name__ == "__main__" else __file__).resolve()
-
-DATE_RE = r"\d{4}-(12|11|10|0?[1-9])-(31|30|[0-2]?\d)T(2[0-3]|1\d|0?[0-9])(:(\d|[0-5]\d)){2}(\.\d{3})?Z"
-
-GroupID = constr(regex=r"^\S+$")
-
-# Pydantic does not support exporting a jsonschema with Dict keys being something else than a str
-# this is a regex for having uuids of type: 8-4-4-4-12 digits
-_NodeIDForDict = constr(
-    regex=r"^[0-9a-fA-F]{8}-?[0-9a-fA-F]{4}-?[0-9a-fA-F]{4}-?[0-9a-fA-F]{4}-?[0-9a-fA-F]{12}$"
-)
 ProjectID = UUID
 ClassifierID = str
-Workbench = Dict[_NodeIDForDict, Node]
-
-
-class WorkbenchUI(BaseModel):
-    position: Position = Field(..., description="The node position in the workbench")
-
-    class Config:
-        extra = Extra.forbid
-
-
-class Slideshow(BaseModel):
-    position: int = Field(..., description="Slide's position", example=["0", "2"])
-
-    class Config:
-        extra = Extra.forbid
-
-
-class StudyUI(BaseModel):
-    workbench: Optional[Dict[_NodeIDForDict, WorkbenchUI]] = Field(None)
-    slideshow: Optional[Dict[_NodeIDForDict, Slideshow]] = Field(None)
-    current_node_id: Optional[NodeID] = Field(alias="currentNodeId")
-
-    class Config:
-        extra = Extra.allow
-
-
-class AccessRights(BaseModel):
-    read: bool = Field(..., description="gives read access")
-    write: bool = Field(..., description="gives write access")
-    delete: bool = Field(..., description="gives deletion rights")
-
-    class Config:
-        extra = Extra.forbid
-
-
-class Owner(BaseModel):
-    user_id: PositiveInt = Field(
-        ...,
-        description="Owner's identifier when registered in the user's database table",
-        example=[2],
-    )
-    first_name: str = Field(..., description="Owner first name", example=["John"])
-    last_name: str = Field(..., description="Owner last name", example=["Smith"])
-
-    class Config:
-        extra = Extra.forbid
 
-
-class ProjectLocked(BaseModel):
-    value: bool = Field(
-        ..., description="True if the project is locked by another user"
-    )
-    owner: Optional[Owner] = Field(None, description="The user that owns the lock")
-
-    class Config:
-        extra = Extra.forbid
-
-
-class ProjectRunningState(BaseModel):
-    value: RunningState = Field(
-        ..., description="The running state of the project", example=["STARTED"]
-    )
-
-    class Config:
-        extra = Extra.forbid
-
-
-class ProjectState(BaseModel):
-    locked: ProjectLocked = Field(..., description="The project lock state")
-    state: ProjectRunningState = Field(..., description="The project running state")
-
-    class Config:
-        extra = Extra.forbid
+# TODO: for some reason class Workbench(BaseModel): __root__= does not work as I thought ... investigate!
+Workbench = Dict[NodeID_AsDictKey, Node]
```

Comment:
I remember having the same problem and came back to this same solution. Please let me know if you figure it out.